llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1460,6 +1460,13 @@ static bool checkAndReplaceCondition(
           checkCondition(Cmp->getPredicate(), Cmp->getOperand(0),
                          Cmp->getOperand(1), Cmp, Info))
     return ReplaceCmpWithConstant(Cmp, *ImpliedCondition);
+
+  if (Cmp->hasSameSign() && Cmp->isRelational())
+    if (auto ImpliedCondition =
+            checkCondition(Cmp->getFlippedSignednessPredicate(),
```

Comment:
Nit: `getSignedPredicate`

Reply:
IMHO getFlippedSignednessPredicate is semantically more correct here, even if in practice it makes no different between samesign typically only occurs on unsigned predicates.

If you want to use getSignedPredicate(), should also check isUnsigned instead of isRelational.